### DISCUSSION TITLE
Consider documenting the fact that the installation process with Conda is long and contains worrying messages such as `solving environment: failed`

### DISCUSSION THREAD

**Discussion by PierreG-activeviam** [atoti/atoti#653] [General]
## Description
On the atoti installation page
https://docs.atoti.io/latest/getting_started/installation.html#conda-package

the command 
_conda install atoti atoti-jupyterlab python_

triggers the following environment resolution problem
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: /

The recommandation is to ensure channels are properly added 
conda config --add channels conda-forge
conda config --add channels https://conda.atoti.io

If the environment resolution problem remains, 
the recommandation to install in 3 steps instead of 1
_conda install python_
_conda install atoti_
_conda install atoti-jupyterlab_

When executing
_conda install atoti_
If the environment resolution problem remains, 
the recommandation is simply to wait 

The root cause for this remains to investigate (it could simply be a server being unavailable during european night time)
Until then, it would be great to have this documented so users know there is no particular action to take, just wait.

**Comment by CamilleBlanco** [comment]
Hello Pierre ! 

Just to be sure I understand: the environment resolution is slow with conda when you install atoti, so you would like to see in the documentation that this is a regular behavior with conda ? 😄

**Comment by PierreG-activeviam** [comment]
Hi Camille, thanks for reaching out. 
Well the waiting time may or may not be documented, as it could become fast in a future version. 
The main thing is to document that the messages starting with _Solving environment: failed...._ are expected.

What happens when you don't know is that, as you are waiting, you google this to see if it's normal.
Google says it's a known issue related to channels addition, and there is a recommended fix. 
At that stage, after a good 5mn waiting, you interrupt the command with CTRL+C, and try to fix the issue, only to see the fix doesn't have any effect, etc...

So, knowing that this message is expected is an important information
especially when the command takes a while to kick start.

Hope this helps
Pierre

**Comment by CamilleBlanco** [comment]
Hello Pierre,

I get the point, many thanks for the details!
We will discuss adding this explanation to the documentation on our side with @tibdex, we will come back to you accordingly.

Thank you for pointing this out!

**Comment by alexis-bvq** [comment]
Hello, following this discussion, I think the slowness is really noticeable since version 0.6.6. When I am installing atoti on clean conda environment it seems way longer than on previous version, especially in order to solve the environment.
As an example it has been 15 minutes I am waiting on the solving environment steps on an aws instance and still no response...
If I try to install only python, pandas or jupyter lab it is way faster, I don't know if it is something you can investigate but I thought it was worth mentioning it.

**Reply by bloussou** [reply]
When you install atoti, you install `pandas` and `jupyter-lab` so it is "normal" that atoti takes more time to be installed since conda is checking version compatibility between all these libraries.

It is a known issue of conda to be slow especially on windows... You can use [mamba](https://github.com/mamba-org/mamba) (we are using it on our CI since conda is too slow) if you want something faster.

to improve speed of package resolution you can remove the defaults channel to only have conda-forge and avoid scanning all the libraries:

```
steps:
      - run: conda config --remove channels defaults
      - run: conda config --add channels conda-forge
```

Once [poetry 1.2.*](https://github.com/python-poetry/poetry) is out it should be faster and cleaner than conda to manage environments.
